### PR TITLE
Tournament Brackets, Manage Events Page, and Edit Teams Page

### PR DESCRIPTION
**Pull request by VoxelPenguin**
### Tournament Brackets
A new tournament bracket component for viewing brackets and selecting match winners.  The component takes in an event ID and generates a bracket based on the teams assigned to that event.

### Edit Events Page
A page for updating the configuration of existing events.  This functions similarly to the Edit Gameboard Space Types page.

### Edit Teams Page
A page for editing the teams of existing events and who's assigned to them.  Event teams are edited locally on the client side, and then changes can be pushed to the backend via save button.  There is a database function for handling all changes to event teams.

### Future Work
The tournament bracket component and Manage Events page need to be incorporated into the overall flow of the session.  Currently, there is a "Create Brackets" link on the GM Tools page for viewing brackets.  Brackets should be viewable for the active event, and potentially for past/future events.  Brackets also need to be locked down such that only the GM can select winners.

### CODE REVIEW

File: src/app/gm-tools/feature/create-bracket-page.component.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    DropdownModule,
+  ],
+})
+export default class CreateBracketPageComponent {
+  private readonly eventService = inject(EventService);
+
+  readonly events = computed(() => this.eventService.events());
+  selectedEvent: Partial<EventModel> = {};
+
+  updateNumberOfTeams(
```
Comment (c-o-l-i-n) [inline src/app/gm-tools/feature/create-bracket-page.component.ts]: Remove unused function

File: src/app/gm-tools/feature/create-bracket-page.component.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        <p-dropdown
+          [options]="events()!"
+          [(ngModel)]="selectedEvent"
+          optionLabel="id"
+          placeholder="Select an Event"
+          styleClass=" w-full"
+        >
+          <ng-template let-ev pTemplate="selectedItem">
+            <div class="flex align-items-center gap-2">
+              <div>{{ ev.name }}</div>
```
Comment (c-o-l-i-n) [inline src/app/gm-tools/feature/create-bracket-page.component.ts]: I assume the purpose of this HTML is to show the event's name in the dropdown rather than its ID or some other data. PrimeNG actually offers a nice way to do this without much code!

https://primeng.org/dropdown#api.dropdown.props.optionLabel

Use `[optionValue]` to specify the field of an `event` that goes into the `[(ngModel)]` (in this case, `id`), and use `[optionLabel]` to specify the field that should show in the dropdown (in this case, `name`).

```html
<p-dropdown
  [options]="events()"
  [(ngModel)]="selectedEvent"
  optionValue="id"
  optionLabel="name"
  placeholder="Select an Event"
  styleClass="w-full"
/>
```

File: src/app/gm-tools/feature/create-bracket-page.component.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      <joshies-header-link
+        text="GM Tools"
+        routerLink=".."
+        chevronDirection="left"
+      />
+    </joshies-page-header>
+    <div class="flex flex-column mt-5 font-semibold">
+      @if (events()) {
+        <p-dropdown
+          [options]="events()!"
```
Comment (c-o-l-i-n) [inline src/app/gm-tools/feature/create-bracket-page.component.ts]: I've found that it's safer to avoid `!` in HTML wherever possible, even if I'm not expecting it to cause an issue. Unhandled `null`/`undefined` values could cause the app to crash.

I see you're already checking `@if (events())`, so you _know_ it's going to be defined. But the Angular compiler isn't smart enough to know that `events()` is always defined inside the `@if` block.

To get around this issue, we can use `as`:

```html
@if (events(); as events) {
  <!-- Now, you can use the value of `events()` with `events`. -->
  <!-- Angular knows that it's defined, so you don't need `!` -->
  <p-dropdown
    [options]="events" 
  />

  <!-- etc. -->
}
```

For a better user experience, I always like to add an `@else` clause that shows a message when the data does not exist. And idk if you remember me mentioning this, but the convention I've used for this app is `undefined` means "the data is still loading", whereas `null` means "it loaded, but there's no data to show"

```html
@else if (events() === null) {
  <!-- Show message when there is no data -->
  No event data available
} @else {
  <!-- Show loading skeleton while loading -->
  <p-skeleton height="2.5rem" />
}
```

File: src/app/gm-tools/feature/create-bracket-page.component.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+          </ng-template>
+          <ng-template let-ev pTemplate="item">
+            <div class="flex align-items-center gap-2">
+              <div>{{ ev.name }}</div>
+            </div>
+          </ng-template>
+        </p-dropdown>
+      }
+    </div>
+    <joshies-tournament-bracket [eventId]="selectedEvent.id!" />
```
Comment (c-o-l-i-n) [inline src/app/gm-tools/feature/create-bracket-page.component.ts]: Similar comment as above: Using `!` is a bit "dangerous," so instead we could wrap this in an `@if` block:

```html
@if (selectedEvent) {
  <joshies-tournament-bracket [eventId]="selectedEvent.id" />
}
```

Angular is smart enough to know that `selectedEvent` is defined inside of the `@if` block. (It just can't do that with Signals yet because Signals are accessed through a function call, which makes type narrowing trickier)

File: src/app/gm-tools/feature/create-bracket-page.component.ts [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  `,
+  styles: ``,
```
Comment (c-o-l-i-n) [inline src/app/gm-tools/feature/create-bracket-page.component.ts]: Remove empty styles (just delete this line)

File: src/app/gm-tools/feature/create-bracket-page.component.ts [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    FormsModule,
+    InputNumberModule,
+    TournamentBracketComponent,
+    DropdownModule,
+  ],
+})
+export default class CreateBracketPageComponent {
+  private readonly eventService = inject(EventService);
+
+  readonly events = computed(() => this.eventService.events());
```
Comment (c-o-l-i-n) [inline src/app/gm-tools/feature/create-bracket-page.component.ts]: Since this computed Signal is just getting the value of another Signal, it doesn't need to be a computed Signal at all:

```typescript
readonly events = this.eventService.events;
```

File: src/app/gm-tools/feature/create-bracket-page.component.ts [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+  selectedEvent: Partial<EventModel> = {};
```
Comment (c-o-l-i-n) [inline src/app/gm-tools/feature/create-bracket-page.component.ts]: I'm not 100% sure why this code uses `Partial`. If the goal is to be "An event, or not an event," we could make the type `EventModel | null`.

We should also probably make this a Signal. It works just fine as a regular variable, but Signals are good for performance. The only potential change you would need to make in the template is to use the "getter function" in the `@if` statement.

```typescript
readonly selectedEvent = signal<EventModel | null>(null);
```

```html
@if (selectedEvent(); as selectedEvent) {
  <joshies-tournament-bracket [eventId]="selectedEvent.id" />
}
```

File: src/app/gm-tools/feature/edit-event-page.component.ts [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+      teamSize: originalEvent.team_size,
+      scoringMap: `{${originalEvent.scoring_map.toString()}}`,
+      imageUrl: originalEvent.image_url ?? '',
+      pointsLabel: originalEvent.points_label ?? 'points',
+      lowerScoresAreBetter: originalEvent.lower_scores_are_better,
+      format:
+        (originalEvent.format as EventFormat) ??
+        EventFormat.ScoreBasedSingleRound,
+    });
+  }
```
Comment (c-o-l-i-n) [inline src/app/gm-tools/feature/edit-event-page.component.ts:130]: Good stuff! 🙌

File: src/app/gm-tools/feature/edit-event-teams-page.component.ts [hunk 9]
```diff
@@ -0,0 +1,610 @@
...
+        (player) =>
+          ({
+            participant_id: -player.player_id,
+            team_id: TeamIds.UnassignedTeam,
+            player_id: player.player_id,
+            display_name: player.display_name,
+            avatar_url: player.avatar_url,
+          }) as EventParticipantWithPlayerInfo,
+      ) ?? ([] as EventParticipantWithPlayerInfo[])),
+    ...(structuredClone(this.databaseEventParticipants()) ??
```
Comment (c-o-l-i-n) [inline src/app/gm-tools/feature/edit-event-teams-page.component.ts:324]: I would move all of this logic for mapping Players to Event Participants into the EventService. I imagine we might eventually need to reuse this in other pages.

File: src/app/gm-tools/feature/edit-event-teams-page.component.ts [hunk 10]
```diff
@@ -0,0 +1,610 @@
...
+      participants: this.localEventParticipants()?.filter(
+        (eventParticipant) => eventParticipant.team_id === eventTeam.id,
+      ),
+    })) ?? []),
+
+    // include a "new team" so the container type matches for drag/drop
+    {
+      ...this.newTeamTemplate(),
+      id: TeamIds.NewTeam,
+    } as EventTeamWithParticipantInfo,
```
Comment (c-o-l-i-n) [inline src/app/gm-tools/feature/edit-event-teams-page.component.ts:350]: Same comment here - we'll probably want to reuse this logic, so it should go in EventService

File: src/app/gm-tools/feature/edit-event-teams-page.component.ts [hunk 11]
```diff
@@ -0,0 +1,610 @@
...
+        previousTeamParticipantCount! <= 1 &&
+        drop.previousContainer.data.id !== TeamIds.UnassignedTeam
+      ) {
+        const previousTeamIndex = eventTeams?.findIndex(
+          (eventTeam) => eventTeam.id === drop.previousContainer.data.id,
+        );
+        eventTeams?.splice(previousTeamIndex!, 1);
+      }
+
+      return [...eventTeams!];
```
Comment (c-o-l-i-n) [inline src/app/gm-tools/feature/edit-event-teams-page.component.ts:482]: Nice job creating immutable Signal updates!

File: src/app/gm-tools/feature/edit-event-teams-page.component.ts [hunk 12]
```diff
@@ -0,0 +1,610 @@
...
+        this.eventService.updateEventTeams(
+          JSON.parse(JSON.stringify(eventTeamUpdates)),
+        ),
+      messageService: this.messageService,
+      confirmationService: this.confirmationService,
+      submittingSignal: this.submitting,
+      successNavigation: null,
+    });
+  }
+}
```
Comment (c-o-l-i-n) [inline src/app/gm-tools/feature/edit-event-teams-page.component.ts:610]: This page is a beast, and you did an awesome job. It feels great to use, and the code is pretty good too 😘

File: src/app/gm-tools/feature/gm-tools-page.component.ts [hunk 13]
```diff
@@ -150,30 +150,37 @@ export default class GmToolsPageComponent {
...
-        text: 'Manage Events',
-        iconClass: 'pi pi-flag bg-purple-500',
-        routerLink: './events',
-      },
-      ...this.sessionStatusDependentLinks[
-        this.gameStateService.sessionStatus() ?? 'undefined'
-      ],
-    ],
+    (): CardLinkModel[] => {
+      return [
```
Comment (c-o-l-i-n) [inline src/app/gm-tools/feature/gm-tools-page.component.ts]: Any reason you changed this from an "expression" arrow function to a "normal" arrow function? I usually like to keep expression arrow functions when possible

File: src/app/gm-tools/feature/lib.routes.ts [hunk 14]
```diff
@@ -185,6 +186,23 @@ const gmToolsRoutes: Routes = [
     loadComponent: () => import('./new-event-page.component'),
     data: { pageAnimationLayer: 2 },
   },
+  {
+    path: 'events/edit/:eventId',
+    loadComponent: () => import('./edit-event-page.component'),
+    data: { pageAnimationLayer: 2 },
+    resolve: { originalEvent: editEventResolver },
```
Comment (c-o-l-i-n) [inline src/app/gm-tools/feature/lib.routes.ts:193]: Look at you using resolvers. They grow up so fast...

File: src/app/shared/data-access/event.service.ts [hunk 15]
```diff
@@ -65,10 +137,79 @@ export class EventService {
...
     );
 
-  readonly events = toSignal(this.events$);
-  readonly eventForThisRound = toSignal(this.eventForThisRound$);
   readonly eventForNextRound = toSignal(this.eventForNextRound$);
 
+  readonly events = toSignal(this.events$);
+
+  readonly eventTeams$: Observable<EventTeamModel[] | null> = this.events$.pipe(
+    map((events) => events?.map((ev) => ev.id)),
```
Comment (c-o-l-i-n) [inline src/app/shared/data-access/event.service.ts]: I believe using optional chaining (`?.`) always returns `undefined`, even if the value is `null`. If that's true, then the `whenNotNull` function wouldn't work how we expect, since the returned value would be `undefined` instead of `null` We could fix it by adding `?? null`:

```typescript
map((events) => events?.map((ev) => ev.id) ?? null),
```

File: src/app/shared/data-access/event.service.ts [hunk 16]
```diff
@@ -65,10 +137,79 @@ export class EventService {
...
+      ...eventTeam,
+      participants: this.eventParticipantsWithPlayerInfo()?.filter(
+        (eventParticipant) => eventParticipant.team_id === eventTeam.id,
+      ),
+    })),
+  );
+
+  readonly eventParticipants$: Observable<EventParticipantModel[] | null> =
+    this.eventTeams$.pipe(
+      map((eventTeams) => eventTeams?.map((eventTeam) => eventTeam.id)),
```
Comment (c-o-l-i-n) [inline src/app/shared/data-access/event.service.ts]: Same comment as above - add `?? null`:

```typescript
map((eventTeams) => eventTeams?.map((eventTeam) => eventTeam.id) ?? null),
```

File: src/app/shared/data-access/event.service.ts [hunk 17]
```diff
@@ -65,10 +137,79 @@ export class EventService {
...
+
+  readonly eventTeamsWithParticipantInfo: Signal<
+    EventTeamWithParticipantInfo[] | undefined
+  > = computed(() =>
+    this.eventTeams()?.map((eventTeam) => ({
+      ...eventTeam,
+      participants: this.eventParticipantsWithPlayerInfo()?.filter(
+        (eventParticipant) => eventParticipant.team_id === eventTeam.id,
+      ),
+    })),
```
Comment (c-o-l-i-n) [inline src/app/shared/data-access/event.service.ts]: To go along with our theme of "undefined means loading" and "null means loaded, but no data", we can modify this Signal to handle both of those cases:

```typescript
readonly eventTeamsWithParticipantInfo: Signal<EventTeamWithParticipantInfo[] | null | undefined
> = computed(() => {
  const eventTeams = this.eventTeams();
  const eventParticipantsWithPlayerInfo = this.eventParticipantsWithPlayerInfo();
  
  // return null or undefined, depending on what the value is
  if (!eventTeams) return eventTeams;
  if (!eventParticipantsWithPlayerInfo) return eventParticipantsWithPlayerInfo;
  
  
  return this.eventTeams.map((eventTeam) => ({
    ...eventTeam,
    participants: eventParticipantsWithPlayerInfo.filter(
      (eventParticipant) => eventParticipant.team_id === eventTeam.id,
    ),
  }));
});
```

File: src/app/shared/ui/tournament-bracket.component.ts [hunk 18]
```diff
@@ -0,0 +1,203 @@
...
+              participant of node.data.participants;
+              track participant.participant_id;
+              let index = $index, last = $last
+            ) {
+              {{
+                (last && index !== 0 ? '& ' : '') +
+                  participant.display_name +
+                  (index < node.data.participants.length - 2 ? ',' : '')
+              }}
+            }
```
Comment (c-o-l-i-n) [inline src/app/shared/ui/tournament-bracket.component.ts]: I would convert this logic into a reusable utility function. Then use that utility function in a Pipe so it can be reused in other pages.

File: src/app/shared/ui/tournament-bracket.component.ts [hunk 19]
```diff
@@ -0,0 +1,203 @@
...
+
+    if (siblingIndex !== -1) {
+      selectedNodes.splice(siblingIndex, 1);
+    }
+  }
+
+  private getNodeParent(
+    node: TreeNode,
+    rootNode: TreeNode,
+  ): TreeNode | undefined {
```
Comment (c-o-l-i-n) [inline src/app/shared/ui/tournament-bracket.component.ts:178]: This is fine, but it's so silly that we need to do this. `parent` always being `undefined` is probably a bug in PrimeNG. Maybe we could file a bug report at some point.

**Review by c-o-l-i-n — CHANGES_REQUESTED** [review]
Great work!! Still room for improvement, but overall superb 👨‍🍳🤌

File: package-lock.json
Comment (c-o-l-i-n) [inline package-lock.json:1]: See if you can revert the changes in `package-lock.json` to what we have in the `main` branch. It seems to be causing the Cloudflare build to fail.

File: src/app/gm-tools/feature/create-bracket-page.component.ts (hunk 1 above)
Comment (VoxelPenguin) [reply src/app/gm-tools/feature/create-bracket-page.component.ts]: Removed.

File: src/app/gm-tools/feature/create-bracket-page.component.ts (hunk 2 above)
Comment (VoxelPenguin) [reply src/app/gm-tools/feature/create-bracket-page.component.ts]: Refactored to use optionValue and optionLabel properties.

File: src/app/gm-tools/feature/create-bracket-page.component.ts (hunk 3 above)
Comment (VoxelPenguin) [reply src/app/gm-tools/feature/create-bracket-page.component.ts]: Refactored to use `@if (events(); as events)` and show placeholder text if an event hasn't been selected.  Also includes a loading skeleton.

File: src/app/gm-tools/feature/create-bracket-page.component.ts (hunk 4 above)
Comment (VoxelPenguin) [reply src/app/gm-tools/feature/create-bracket-page.component.ts]: Refactored to use an `@if` block.  `selectedEvent` is now also `selectedEventId` of type `number | null` because the bracket component only needs an `eventId` as input.

File: src/app/gm-tools/feature/create-bracket-page.component.ts (hunk 5 above)
Comment (VoxelPenguin) [reply src/app/gm-tools/feature/create-bracket-page.component.ts]: Oops lol.  Deleted.

File: src/app/shared/data-access/event.service.ts (hunk 15 above)
Comment (VoxelPenguin) [reply src/app/shared/data-access/event.service.ts]: Makes sense.  Added `?? null`.

File: src/app/shared/data-access/event.service.ts (hunk 16 above)
Comment (VoxelPenguin) [reply src/app/shared/data-access/event.service.ts]: Added `?? null`.

File: src/app/gm-tools/feature/create-bracket-page.component.ts (hunk 6 above)
Comment (VoxelPenguin) [reply src/app/gm-tools/feature/create-bracket-page.component.ts]: Yep, not sure what I was thinking here.  Updated.

File: src/app/gm-tools/feature/create-bracket-page.component.ts (hunk 7 above)
Comment (VoxelPenguin) [reply src/app/gm-tools/feature/create-bracket-page.component.ts]: I updated `selectedEvent` to be `selectedEventId: number | null`.  This is simpler since the bracket component only needs an `eventId` as input.

File: src/app/gm-tools/feature/edit-event-page.component.ts (hunk 8 above)
Comment (VoxelPenguin) [reply src/app/gm-tools/feature/edit-event-page.component.ts:130]: Thanks!

File: src/app/gm-tools/feature/edit-event-teams-page.component.ts (hunk 11 above)
Comment (VoxelPenguin) [reply src/app/gm-tools/feature/edit-event-teams-page.component.ts:482]: Thanks!  They're as immutable as my spirit is indomitable.

File: src/app/gm-tools/feature/create-bracket-page.component.ts (hunk 7 above)
Comment (c-o-l-i-n) [reply src/app/gm-tools/feature/create-bracket-page.component.ts]: Can you make it a signal too?


```typescript
readonly selectedEventId = signal<number | null>(null);
```

File: src/app/gm-tools/feature/edit-event-teams-page.component.ts (hunk 12 above)
Comment (VoxelPenguin) [reply src/app/gm-tools/feature/edit-event-teams-page.component.ts:610]: :kissing_heart::kissing_heart::kissing_heart:

File: src/app/gm-tools/feature/edit-event-teams-page.component.ts (hunk 11 above)
Comment (c-o-l-i-n) [reply src/app/gm-tools/feature/edit-event-teams-page.component.ts:482]: I had to google what that meant

File: src/app/gm-tools/feature/gm-tools-page.component.ts (hunk 13 above)
Comment (VoxelPenguin) [reply src/app/gm-tools/feature/gm-tools-page.component.ts]: I remember this happened when I rebased awhile back.  I didn't look too closely at it because I thought it was coming from main.  Not sure if I just looked at it wrong or what, but it's been changed back.

File: src/app/gm-tools/feature/lib.routes.ts (hunk 14 above)
Comment (VoxelPenguin) [reply src/app/gm-tools/feature/lib.routes.ts:193]: Don't worry, I'll also be your little boy Dad!

Comment (c-o-l-i-n) [reply src/app/gm-tools/feature/lib.routes.ts:193]: S-tier PR comment

File: src/app/gm-tools/feature/edit-event-teams-page.component.ts (hunk 9 above)
Comment (VoxelPenguin) [reply src/app/gm-tools/feature/edit-event-teams-page.component.ts:324]: I'm already mapping player info to participant info in the event service as `eventParticipantsWithPlayerInfo`.  I'm assuming you're referring to the below block of code, but all this does is create some "unassigned" participants to make it easier to drag/drop and determine which players aren't on a team after the edits have been made.

``` ts
.map(
        (player) =>
          ({
            participant_id: -player.player_id,
            team_id: TeamIds.UnassignedTeam,
            player_id: player.player_id,
            display_name: player.display_name,
            avatar_url: player.avatar_url,
          }) as EventParticipantWithPlayerInfo,
      ) ?? ([] as EventParticipantWithPlayerInfo[])
```

File: src/app/shared/ui/tournament-bracket.component.ts (hunk 19 above)
Comment (VoxelPenguin) [reply src/app/shared/ui/tournament-bracket.component.ts:178]: Agreed, I tried a few different ways of assigning `parent` with no luck.  I can't say I'm 100% certain I wasn't doing something wrong, but I'm pretty sure this is a bug.

File: src/app/gm-tools/feature/edit-event-teams-page.component.ts (hunk 10 above)
Comment (VoxelPenguin) [reply src/app/gm-tools/feature/edit-event-teams-page.component.ts:350]: My only issue with this is that `eventTeamsWithParticipantInfo` gets its value from local changes (i.e. `localSortedEventTeams`) and includes two extra "teams": one for unassigned players and one for creating new teams.  That way, t serves as a "view model" for the HTML to make it easier to drag/drop and show event participants with their associated teams.  So we couldn't just create this in the event service as-is because it has some extra stuff specific to the Edit Teams page.

I think the best approach would be to add `eventTeamsWithParticipantInfo` to the event service, change `databaseEventTeams` and `localSortedEventTeams` to get their value from this signal in event service, and continue to use the view model the same way.

File: src/app/shared/data-access/event.service.ts (hunk 17 above)
Comment (VoxelPenguin) [reply src/app/shared/data-access/event.service.ts]: ~~Agreed, I'll add this when I put the Signal into the Event Service.~~

Oops, I'm forgetting this already _is_ in the Event Service.  This is updated in Event Service.

File: package-lock.json
Comment (VoxelPenguin) [reply package-lock.json:1]: Yay, sounds fun.

Comment (c-o-l-i-n) [reply package-lock.json:1]: You could probably just copy and paste what we have in main

File: src/app/gm-tools/feature/edit-event-teams-page.component.ts (hunk 10 above)
Comment (c-o-l-i-n) [reply src/app/gm-tools/feature/edit-event-teams-page.component.ts:350]: Gotcha. I didn’t take the time to fully digest what was going on here. Do whatever you think is best 😁

File: package-lock.json
Comment (VoxelPenguin) [reply package-lock.json:1]: Yeah I'll just do that.  I ran `npm upgrade` yesterday and then when I rebased I used all the newer versions... probably a mistake...

File: src/app/gm-tools/feature/create-bracket-page.component.ts (hunk 7 above)
Comment (VoxelPenguin) [reply src/app/gm-tools/feature/create-bracket-page.component.ts]: Oops, yes it's a signal now.

File: src/app/gm-tools/feature/edit-event-teams-page.component.ts (hunk 10 above)
Comment (VoxelPenguin) [reply src/app/gm-tools/feature/edit-event-teams-page.component.ts:350]: I ended up leaving it how it was, it was causing more problems than it was worth to change the `WritableSignal` version of this on the Edit Teams page.

File: src/app/shared/ui/tournament-bracket.component.ts (hunk 18 above)
Comment (VoxelPenguin) [reply src/app/shared/ui/tournament-bracket.component.ts]: Created `event-helpers.ts` and `participant-list.pipe.ts` to implement this logic as a pipe.

File: package-lock.json
Comment (VoxelPenguin) [reply package-lock.json:1]: Copied the current `package-lock.json` file from main.

**Review by c-o-l-i-n — APPROVED** [review]
